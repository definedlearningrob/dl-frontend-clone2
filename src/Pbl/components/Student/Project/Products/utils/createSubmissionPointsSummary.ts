export const createSubmissionPointsSummary = (pointsAvailable: number, pointsScored: number) =>
  `${pointsScored}/${pointsAvailable}`;
