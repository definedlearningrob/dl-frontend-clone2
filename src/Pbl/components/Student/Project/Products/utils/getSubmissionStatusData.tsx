import { capitalize } from 'lodash-es';

import { PRODUCT_SUBMISSION_STATUS } from '@pbl/resources/enums';

import { SUBMISSION_STATUS } from '@shared/components/SubmissionStatusBadge';

export const getSubmissionStatusData = (
  status: PRODUCT_SUBMISSION_STATUS,
  gradePoints?: string | null
) => {
  const statusLabel = capitalize(status).replace(/[_]/g, ' ');
  const label = [statusLabel, gradePoints].filter(Boolean).join(' ');

  switch (status) {
    case PRODUCT_SUBMISSION_STATUS.DRAFT:
      return { label, value: SUBMISSION_STATUS.DRAFT };
    case PRODUCT_SUBMISSION_STATUS.SUBMITTED:
      return {
        label,
        value: SUBMISSION_STATUS.SUBMITTED,
      };
    case PRODUCT_SUBMISSION_STATUS.GRADED:
      return { label, value: SUBMISSION_STATUS.GRADED };
    case PRODUCT_SUBMISSION_STATUS.NOT_STARTED:
      return {
        label,
        value: SUBMISSION_STATUS.NOT_STARTED,
      };
  }
};
