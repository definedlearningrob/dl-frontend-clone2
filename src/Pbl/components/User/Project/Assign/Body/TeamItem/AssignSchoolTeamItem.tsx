import { ChangeEvent, useEffect, useState } from 'react';
import { isEmpty } from 'lodash-es';
import { useField } from 'formik';

import { type TSchoolClass } from '@pbl/graphql/user/queries/schoolClasses';

import SharedCheckbox from '@shared/components/Checkbox/Checkbox';
import DeprecatedIconButton from '@shared/components/DeprecatedIconButton/DeprecatedIconButton';
import { ReactComponent as ChevronDownIcon } from '@shared/svg/chevron_down.svg';
import { useFilterContext } from '@shared/hooks/useFilterContext';

import styles from '../ProjectAssignBody.module.sass';
import { hasIncludedStudent, hasIncludedTeam } from '../helpers';

import { TeamItem } from './TeamItem';

type Props = {
  schoolClass: TSchoolClass;
};

export const ProjectAssignSchoolTeamItem = ({ schoolClass }: Props) => {
  const [isOpen, setIsOpen] = useState(false);
  const [teamIdsField, _, fieldHelpers] = useField('teamIds');
  //@ts-ignore
  const { filter } = useFilterContext();

  const teams = schoolClass.teams;
  const students = schoolClass.students.nodes;
  const teamIds = teams.map((team) => team.uuid);
  const formikTeamIds = teamIdsField.value || [];

  const hasSomeTeamsEnrolled = formikTeamIds.some((teamId: string) => teamIds.includes(teamId));

  const hasAllTeamsEnrolled =
    !isEmpty(teams) && teamIds.every((teamId: string) => formikTeamIds.includes(teamId));

  const indeterminate = !hasAllTeamsEnrolled && hasSomeTeamsEnrolled;

  useEffect(() => {
    if (!isOpen && filter.nameCont) {
      const searchedName = filter.nameCont.toLowerCase();

      if (hasIncludedStudent(students, searchedName) || hasIncludedTeam(teams, searchedName)) {
        setIsOpen(true);
      }
    }
  }, [filter.nameCont]);

  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    const newValue = event.target.checked;

    if (newValue && !indeterminate) {
      const uniqueNewTeamIds = [...new Set([...formikTeamIds, ...teamIds])];

      fieldHelpers.setValue(uniqueNewTeamIds);
    } else {
      fieldHelpers.setValue(formikTeamIds.filter((teamId: string) => !teamIds.includes(teamId)));
    }
  };

  return (
    <li key={schoolClass.uuid}>
      <div className={styles.listItem}>
        <DeprecatedIconButton
          className={styles.checkbox}
          icon={<ChevronDownIcon />}
          size='sm'
          onClick={() => setIsOpen(!isOpen)}
        />
        <SharedCheckbox
          checked={hasAllTeamsEnrolled}
          className={styles.checkbox}
          id={schoolClass.uuid}
          indeterminate={indeterminate}
          onChange={onChange}
        />
        <label htmlFor={schoolClass.uuid}>{schoolClass.name}</label>
      </div>
      {isOpen && !isEmpty(schoolClass.teams) && (
        <ul className={styles.unorderedList}>
          {schoolClass.teams.map((team) => (
            <TeamItem key={team.uuid} team={team} />
          ))}
        </ul>
      )}
    </li>
  );
};
