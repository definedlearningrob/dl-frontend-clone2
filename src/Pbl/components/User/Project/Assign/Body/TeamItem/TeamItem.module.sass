@import '@shared/stylesheets/config.sass'

.listItem
  +flex-container(column)
  margin-bottom: $m-sm + $m-xxs

  svg
    color: $color-primary-500

.checkbox
  display: inline-block
  margin: 0
  padding: 0

.listContent
  +flex-container(row, flex-start, center)

.students
  font-size: $font-size-sm
  font-weight: $font-weight-regular
  line-height: $line-height-lg
  margin-left: $m-lg
  margin-top: $m-xs

.studentItem:after
  content: ','
  margin-right: $m-xs

.studentItem:last-of-type:after
  content: none
