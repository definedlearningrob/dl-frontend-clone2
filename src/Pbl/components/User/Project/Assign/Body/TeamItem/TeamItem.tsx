import { FormEvent, useEffect, useState } from 'react';
import { isEmpty } from 'lodash-es';
import { useField } from 'formik';

import { TTeam } from '@pbl/graphql/user/queries/schoolClasses';

import SharedCheckbox from '@shared/components/Checkbox/Checkbox';
import DeprecatedIconButton from '@shared/components/DeprecatedIconButton/DeprecatedIconButton';
import { ReactComponent as ChevronDownIcon } from '@shared/svg/chevron_down.svg';
import { useFilterContext } from '@shared/hooks/useFilterContext';

import { hasIncludedStudentsInTeam } from '../helpers';

import styles from './TeamItem.module.sass';

type Props = {
  team: TTeam;
};

export const TeamItem = ({ team }: Props) => {
  //@ts-ignore
  const { filter } = useFilterContext();
  const [isOpen, setIsOpen] = useState(false);
  const [teamIdsField, _, fieldHelpers] = useField('teamIds');

  const checked = (teamId: string) =>
    teamIdsField.value.some((formikTeam: string) => formikTeam === teamId);

  const onChange = (event: FormEvent<HTMLInputElement>, passedTeamId: string) => {
    // @ts-ignore
    const newValue = event.target.checked;

    if (newValue) {
      fieldHelpers.setValue([...teamIdsField.value, passedTeamId]);
    } else {
      fieldHelpers.setValue(teamIdsField.value.filter((teamId: string) => teamId !== passedTeamId));
    }
  };

  useEffect(() => {
    if (!isOpen && filter.nameCont) {
      const searchedName = filter.nameCont.toLowerCase();

      if (hasIncludedStudentsInTeam(team, searchedName)) {
        setIsOpen(true);
      }
    }
  }, [filter.nameCont]);

  return (
    <li key={team.uuid} className={styles.listItem}>
      <div className={styles.listContent}>
        <DeprecatedIconButton
          className={styles.checkbox}
          icon={<ChevronDownIcon />}
          size='sm'
          onClick={() => setIsOpen(!isOpen)}
        />
        <SharedCheckbox
          checked={checked(team.uuid)}
          className={styles.checkbox}
          id={team.uuid}
          onChange={(event) => onChange(event, team.uuid)}
        />
        <label htmlFor={team.uuid}>{team.name}</label>
      </div>
      {isOpen && (
        <div className={styles.students}>
          {!isEmpty(team.students.nodes) &&
            team.students.nodes.map((student) => (
              <span
                key={student.uuid}
                className={styles.studentItem}>{`${student.firstName} ${student.lastName}`}</span>
            ))}
        </div>
      )}
    </li>
  );
};
