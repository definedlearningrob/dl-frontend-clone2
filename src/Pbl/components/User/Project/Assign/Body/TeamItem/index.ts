export * from './AssignSchoolTeamItem';
export * from './TeamItem';
