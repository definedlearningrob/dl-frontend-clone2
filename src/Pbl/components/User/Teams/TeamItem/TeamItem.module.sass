@import '@shared/stylesheets/config.sass'

.team
  :global(.dropdown-portal-container)
    width: 0
    transition: width .1s ease-out

  &:hover
    :global(.dropdown-portal-options__trigger-icon-wrapper)
      background-color: $color-white

    .teamIconWrapper
      background-color: $color-white

    :global(.dropdown-portal-container)
      width: $m-md
