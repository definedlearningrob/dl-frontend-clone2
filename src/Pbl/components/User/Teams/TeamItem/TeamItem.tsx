import { useTranslation } from 'react-i18next';
import cx from 'classnames';

import { TeamAvatars } from '@pbl/components/User/Teams/TeamAvatars/TeamAvatars';
import type { TTeam } from '@pbl/graphql/user/fragments/team';

import { ReactComponent as EditIcon } from '@shared/svg/edit.svg';
import { ReactComponent as DeleteIcon } from '@shared/svg/delete_outlined.svg';
import Dropdown from '@shared/components/Dropdown/Dropdown';
import { TeamHeader } from '@shared/components/TeamHeader';
import { IconContainer } from '@shared/components/IconContainer/IconContainer';

import { ModalTypes } from '../TeamModal';

import styles from './TeamItem.module.sass';

type Props = {
  team: TTeam;
  handleTeamModalOpen: (type: ModalTypes, team: TTeam) => void;
};

export const TeamItem = ({ handleTeamModalOpen, team }: Props) => {
  const { t } = useTranslation();

  const handleEditModalOpen = () => {
    handleTeamModalOpen(ModalTypes.EDIT, team);
  };

  const handleDeleteModalOpen = () => {
    handleTeamModalOpen(ModalTypes.DELETE, team);
  };

  const teamWrapperClassnames = cx(
    styles.team,
    'flex items-center relative font-bold overflow-hidden py-x px-xs xxxl:p-sm hover:bg-primary-200',
    'border-b border-neutral-300 last:border-b-0'
  );

  return (
    <div className={teamWrapperClassnames}>
      <TeamHeader memberCount={team.students.nodesCount} teamName={team.name} />
      <div className='flex items-center absolute right-xs gap-xs'>
        <TeamAvatars teamId={team.uuid} users={team.students.nodes} />
        <Dropdown>
          <Dropdown.Dropdown>
            <Dropdown.Trigger />
            <Dropdown.Options>
              <Dropdown.Option onClick={handleEditModalOpen}>
                <IconContainer Icon={EditIcon} paddingSize='none' size='sm' />
                {t('common.actions.edit')}
              </Dropdown.Option>
              <Dropdown.Option className='!text-danger-500' onClick={handleDeleteModalOpen}>
                <IconContainer Icon={DeleteIcon} paddingSize='none' size='sm' />
                {t('common.actions.delete')}
              </Dropdown.Option>
            </Dropdown.Options>
          </Dropdown.Dropdown>
        </Dropdown>
      </div>
    </div>
  );
};
