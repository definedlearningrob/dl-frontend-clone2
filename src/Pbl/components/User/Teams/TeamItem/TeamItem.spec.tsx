import { screen } from '@testing-library/react';

import { renderWithI18N } from '@pbl/utils/test';
import { TeamItem } from '@pbl/components/User/Teams/TeamItem/TeamItem';

const team = {
  id: '1',
  uuid: 'uuid1',
  name: 'team name',
  isArchived: false,
  students: {
    nodes: [
      {
        currentTasksCount: 1,
        firstName: 'Test',
        lastName: 'Student',
        uuid: '1',
      },
      {
        currentTasksCount: 1,
        firstName: 'Test',
        lastName: 'Student 2',
        uuid: '2',
      },
    ],
    nodesCount: 2,
    pagesCount: 1,
  },
  tasks: [{ displayName: 'task', id: '1' }],
};
describe('TeamItem', () => {
  it('should render correctly', () => {
    const { container } = renderWithI18N(<TeamItem handleTeamModalOpen={jest.fn()} team={team} />);

    expect(screen.getByText('team name')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
