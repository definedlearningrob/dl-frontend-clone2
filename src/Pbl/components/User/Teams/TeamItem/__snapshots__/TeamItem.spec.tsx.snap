// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamItem should render correctly 1`] = `
<div>
  <div
    class="team flex items-center relative font-bold overflow-hidden py-x px-xs xxxl:p-sm hover:bg-primary-200 border-b border-neutral-300 last:border-b-0"
  >
    <div
      class="wrapper"
    >
      <div
        class="teamIconWrapper"
      >
        <div
          class="icon teamIcon"
          data-testid="icon"
        >
          <div />
        </div>
      </div>
      <div>
        <p
          class="memberCount"
        >
          2 members
        </p>
        <h6
          class="teamName"
        >
          team name
        </h6>
      </div>
    </div>
    <div
      class="flex items-center absolute right-xs gap-xs"
    >
      <div
        class="group"
      >
        <div
          class="avatarWrapper"
        >
          <button
            aria-controls="radix-0"
            aria-expanded="false"
            aria-haspopup="dialog"
            data-state="closed"
            type="button"
          >
            <div
              class="avatar"
              data-testid="avatar-image"
            >
              <div
                class="avatar__image sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar__image sb-avatar__text"
                  style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </button>
        </div>
        <div
          class="avatarWrapper"
        >
          <button
            aria-controls="radix-1"
            aria-expanded="false"
            aria-haspopup="dialog"
            data-state="closed"
            type="button"
          >
            <div
              class="avatar"
              data-testid="avatar-image"
            >
              <div
                class="avatar__image sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar__image sb-avatar__text"
                  style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="dropdown-portal-container"
      >
        <div
          class="dropdown-portal-options__trigger"
          data-testid="dropdown-trigger-container"
        >
          <button
            class="dropdown-portal-options__trigger-icon-wrapper"
          >
            <div
              class="icon dropdown-portal-options__trigger-icon -sm-size"
              data-testid="icon"
            >
              <div />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
